lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+
+                 removeDialog: function(invokingElement) {
+                     /* Removes the Bootstrap v4 dialog modal from the update session enrollment button. */
+                     invokingElement.popover('dispose');
+                 },
+
+                 showDialog: function(invokingElement) {
+                     /* Given an element with an associated dialog modal, shows the modal. */
+                     invokingElement.popover('show');
+                 },
+
+                 hideDialog: function(invokingElement) {
+                     /* Hides the  modal without removing it from the DOM. */
+                     invokingElement.focus().popover('hide');
+                 },
+
+                 handleVerificationPopoverA11y: function(e) {
+                     /* Ensure that the second step verification popover is treated as an a11y compliant dialog */
+                     var nextButton,
+                         openButton = $(e.target).closest('.course-entitlement-selection-container')
+                             .find('.enroll-btn-initial');
+                     if (e.key === 'Tab') {
+                         e.preventDefault();
+                         nextButton = $(e.target).is(':first-child') ? $(e.target).next('.final-confirmation-btn') :
+                             $(e.target).prev('.final-confirmation-btn');
+                         nextButton.focus();
+                     } else if (e.key === 'Escape') {
+                         openButton.popover('hide');
+                         openButton.focus();
+                     }
+                 },
+
+                 formatCourseHomeUrl: function(sessionKey) {
+                     /*
+                     Takes the base course home URL and updates it with the new session id, leveraging the
+                     the fact that all course keys contain a '+' symbol.
+                     */
+                     var oldSessionKey = this.courseHomeUrl.split('/')
+                         .filter(
+                             function(urlParam) {
+                                 return urlParam.indexOf('+') > 0;
+                             }
+                         )[0];
+                     return this.courseHomeUrl.replace(oldSessionKey, sessionKey);
+                 },
+
+                 formatDates: function(sessionData) {
+                     /*
+                     Takes a data object containing the upcoming available sessions for an entitlement and returns
+                     the object with a session_dates attribute representing a formatted date string that highlights
+                     the start and end dates of the particular session.
+                      */
+                     var formattedSessionData = sessionData,
+                         startDate,
+                         endDate,
+                         dateFormat;
+                     moment.locale(window.navigator.userLanguage || window.navigator.language);
```

Comment:
I think we should probably use the language of the page, as defined in the HTML lang attribute, as this should be the language that the rest of the page content is rendered in.

Reply:
Fixed